components/micro-gateway-core/src/main/ballerina/gateway/filters/validation_filter.bal

```diff
@@ -82,26 +81,29 @@ public type ValidationFilter object {
                     if (requestPath == i) {
                         json parameters = swagger[PATHS][i][requestMethod][PARAMETERS];
                         //go through each item in parameters array and find the schema property
-                        foreach var k in parameters {
-                            if (k[SCHEMA] != null) {
-                                if (k[SCHEMA][REFERENCE] != null)  {
-                                    //getting the reference to the model
-                                    string modelReference = k[SCHEMA][REFERENCE].toString();
-                                    //getting the model name
-                                    modelName = untaint replaceModelPrefix(modelReference);
-                                    //check whether there is a model available from the asigned model name
-                                    if (models[modelName] != null) {
-                                        model = models[modelName];
-                                    }
-                                } else {
-                                    //getting inline model
-                                    model = k[SCHEMA];
-                                }
-                            }
+                        foreach var (key, val) in check map<json>.convert(parameters) {
+                            // Todo: Fix the Loop
+
+                            // if (v != null) {
+                            //     if (k[SCHEMA][REFERENCE] != null)  {
+                            //         //getting the reference to the model
+                            //         string modelReference = k[SCHEMA][REFERENCE].toString();
+                            //         //getting the model name
+                            //         modelName = untaint replaceModelPrefix(modelReference);
+                            //         //check whether there is a model available from the asigned model name
+                            //         if (models[modelName] != null) {
+                            //             model = models[modelName];
+                            //         }
+                            //     } else {
+                            //         //getting inline model
+                            //         model = k[SCHEMA];
+                            //     }
+                            // }
                         }
```

Comment:
Why this part is commented out

Reply:
This particular logic needs to be refactored analyzing the json structure